website/build_site.sh

```diff
@@ -1,37 +1,117 @@
-#!/bin/sh
+#!/bin/bash
+#
+# A simple script to build the github IO pages out of
+
+# be strict
+set -eu
+
+# be verbose
+#set -x
 
 # ensure we're in the right folder
 scriptdir=$(readlink -f "$(dirname "$0")")
-cd "$scriptdir"
+MLOS_ROOT=$(readlink -f "$scriptdir/..")
+cd "$MLOS_ROOT/website"
 
-cp -r ../documentation content/
-# downgrade html output because hugo doesn't like raw html
-jupyter nbconvert ../source/Mlos.Notebooks/*.ipynb --to markdown --output-dir content/notebooks --template nbconvert_template.md.j2 --config jupyter_nbconvert_config.py
+. "$MLOS_ROOT/scripts/util.sh"
+pythonCmd=$(getPythonCmd)
 
-# nbconvert and hugo disagree about paths
-# this should probably be done via the template
-sed -i 's/BayesianOptimization_files/\.\.\/BayesianOptimization_files/g' content/notebooks/BayesianOptimization.md
-sed -i 's/SmartCacheOptimization_files/\.\.\/SmartCacheOptimization_files/g' content/notebooks/SmartCacheOptimization.md
+# Make sure we have all the requirements for generating notebooks.
+$pythonCmd -m pip install -e "$MLOS_ROOT/source/Mlos.Python/"
+$pythonCmd -m pip install pyyaml
+$pythonCmd -m pip install \
+    $("$MLOS_ROOT/scripts/parse-pip-requirements-from-environment-yaml.py" "$MLOS_ROOT/source/Mlos.Notebooks/environment.yml")
```

Comment:
Possibly.  I was trying to reduce the amount of changes in one already large PR.

Reply:
Part of this btw, was my intention to reuse this in the `Dockerfile` to avoid using `requirements.txt` so we could drop it later on (as discussed).  But I agree, not super elegant.

Reply:
I agree, this PR is already big. Sorry to make you clean up after me ;)